osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -190,32 +201,23 @@ public void handleSsldPendingFirmwareUpdate(final String deviceIdentification) {
         }
     }
 
-    private SsldPendingFirmwareUpdate getMostRecentDeletingOutdatedPendingFirmwareUpdates(
+    private Optional<SsldPendingFirmwareUpdate> getMostRecentSsldPendingFirmwareUpdate(
             final List<SsldPendingFirmwareUpdate> ssldPendingFirmwareUpdates) {
 
-        if (CollectionUtils.isEmpty(ssldPendingFirmwareUpdates)) {
-            throw new IllegalArgumentException("ssldPendingFirmwareUpdates must not be empty");
-        }
-        if (ssldPendingFirmwareUpdates.size() == 1) {
-            return ssldPendingFirmwareUpdates.get(0);
-        }
-
-        LOGGER.warn("Found multiple pending firmware update records for SSLD: {}", ssldPendingFirmwareUpdates);
-
-        final SsldPendingFirmwareUpdate mostRecentSsldPendingFirmwareUpdate = ssldPendingFirmwareUpdates.stream()
+        return ssldPendingFirmwareUpdates.stream()
                 .max(Comparator.comparing(SsldPendingFirmwareUpdate::getCreationTime)
-                        .thenComparing(Comparator.comparing(SsldPendingFirmwareUpdate::getId)))
-                .orElseThrow(() -> new AssertionError("No most recent pending firmware update from a non-empty list"));
+                        .thenComparing(Comparator.comparing(SsldPendingFirmwareUpdate::getId)));
+    }
 
-        ssldPendingFirmwareUpdates.forEach(pendingUpdate -> {
-            if (mostRecentSsldPendingFirmwareUpdate.getId() != pendingUpdate.getId()) {
-                LOGGER.warn("Deleting pending firmware update assumed outdated: {}", pendingUpdate);
-                this.ssldPendingFirmwareUpdateRepository.delete(pendingUpdate);
-            }
-        });
+    private void deleteOutdatedSsldPendingFirmareUpdates(final List<SsldPendingFirmwareUpdate> updatesToDelete,
```

Comment:
Fix typo (missing "w" in firmware):
`deleteOutdatedSsldPendingFirmareUpdates` -> `deleteOutdatedSsldPendingFirmwareUpdates`

Reply:
Added the w.